### PR TITLE
feat: added a simple model that is able to parse git url with optiona…

### PR DESCRIPTION
**Pull request by Zhongtang**
…l branch and path

### CODE REVIEW

File: model/giturl.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+func (git *GitUrl) GetCloneInfo() (url, branch string) {
+	return fmt.Sprintf("git@%+s:%+s/%+s.git", git.Host, git.Org, git.Repo), git.Branch
+}
+
+func New(gitUrlStr string) (*GitUrl, error) {
+	// This would match something like git@github.com:org/repo.git/path#branch
+	// path and branch are optional. If not given, default values are "" and "master"
+	gitUrlRegex, _ := regexp.Compile("^git@([^/:#]+):([^/:#]+)/+([^/:#]+)\\.git(/[^#]*)?(#.+)?")
```
Comment (Filbird) [inline model/giturl.go]: **EDIT:** Never mind, I did not realize that the argument is treated as an interpreted string literal. 

I suppose an alternative is to use raw string literals, \` \`, if we want to use a conventional regex.

----
I believe this regex may be invalid according to regexr.

Here's an alternative that may work:
`^git@([^:]+):([^/]+)\/([^#/]+)(?:\.git)?\/?([^#]*?)(#.+)?$`

https://regexr.com/3p9fp

File: model/giturl.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	if parseResult == nil {
+		return nil, fmt.Errorf("Error: not a valid git url %+s", gitUrlStr)
+	}
+
+	gitUrl := GitUrl{
+		Host:   parseResult[1],
+		Org:    parseResult[2],
+		Repo:   parseResult[3],
+		Branch: parseResult[5],
+		Path:   parseResult[4],
```
Comment (Filbird) [inline model/giturl.go]: Can we swap `Branch:` and `Path:`? 😛

File: model/giturl.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	}
+
+	if gitUrl.Branch == "" {
+		gitUrl.Branch = "master"
+	} else {
+		gitUrl.Branch = strings.TrimPrefix(gitUrl.Branch, "#")
+	}
+
+	if gitUrl.Path != "" {
```
Comment (Filbird) [inline model/giturl.go]: Should we default to `default.xml` here or does it make more sense to do so in the upcoming `repo` package?

File: model/giturl.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
+package model
```
Comment (Filbird) [inline model/giturl.go]: Would it be better conventionally to name this package `git`?

**Review by Filbird — APPROVED** [review]

File: model/giturl.go (hunk 3 above)
Comment (Zhongtang) [reply model/giturl.go]: Let's do this in whoever invokes this function instead, in case we want to reuse it elsewhere

**Review by minzcmu — APPROVED** [review]
Logic LGTM 👍 better have a PR build running! lol

**Pull request merged**